### PR TITLE
feat: introduce `model_kwargs` in Sparse Embedders (can be used for BM25 parameters)

### LINKED ISSUE #1096: add support for bm25 in FastEmbed integration
**Is your feature request related to a problem? Please describe.**
Qdrant introduced support for bm25 on FastEmbed, would need to add some parameters to do tuning.

**Describe the solution you'd like**
Add the parameters for a `bm25` variable into a dictionary and check if the model is `"Qdrant/bm25"` otherwise initialize the variable `bm25` to `None`

**Describe alternatives you've considered**
create a dedicated object for bm25 in the FastEmbed integration

### PR DESCRIPTION
**Pull request by alperkaya**
### Related Issues

- fixes #1096

### Proposed Changes:

 
if qdrant/bm25 model is chosen, there is a new config available to do fine tuning

### How did you test it?

CI, added new unit tests

### Checklist

- I have read the [contributors guidelines](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CODE_OF_CONDUCT.md)
- I have updated the related issue with new insights and changes
- I added unit tests and updated the docstrings
- I've used one of the [conventional commit types](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title: `fix:`, `feat:`, `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`.

### CODE REVIEW

**Comment by anakin87** [comment]
Hello and thank you!

Currently, I see that depending on the sparse technique used, FastEmbed has different config parameters.
- [BM25](https://github.com/qdrant/fastembed/blob/e2e1f93685acc4b64ad30516c8750b4c693a1f14/fastembed/sparse/bm25.py#L101): `k`, `b`, ...
- [BM42](https://github.com/qdrant/fastembed/blob/e2e1f93685acc4b64ad30516c8750b4c693a1f14/fastembed/sparse/bm42.py#L63): `alpha`

So I think it would be better to implement a more agnostic support for model configuration.
Something like `model_kwargs: Optional[Dict[str, Any]] = None`.

If the user provides these configurations, we should try to apply them without additional checks and fail if the chosen model/technique does not support them.

Could you modify the implementation to go in this direction?

**Comment by alperkaya** [comment]
> Hello and thank you!
> 
> Currently, I see that depending on the sparse technique used, FastEmbed has different config parameters.
> 
>     * [BM25](https://github.com/qdrant/fastembed/blob/e2e1f93685acc4b64ad30516c8750b4c693a1f14/fastembed/sparse/bm25.py#L101): `k`, `b`, ...
> 
>     * [BM42](https://github.com/qdrant/fastembed/blob/e2e1f93685acc4b64ad30516c8750b4c693a1f14/fastembed/sparse/bm42.py#L63): `alpha`
> 
> 
> So I think it would be better to implement a more agnostic support for model configuration. Something like `model_kwargs: Optional[Dict[str, Any]] = None`.
> 
> If the user provides these configurations, we should try to apply them without additional checks and fail if the chosen model/technique does not support them.
> 
> Could you modify the implementation to go in this direction?

Hello,

I applied more agnostic solution for model config params. LMK for any more feedback.

Thank you!

File: integrations/fastembed/src/haystack_integrations/components/embedders/fastembed/embedding_backend/fastembed_backend.py [hunk 1]
```diff
@@ -73,14 +73,19 @@ def get_embedding_backend(
         cache_dir: Optional[str] = None,
         threads: Optional[int] = None,
         local_files_only: bool = False,
+        model_kwargs: Optional[Dict[str, Any]] = None,
     ):
-        embedding_backend_id = f"{model_name}{cache_dir}{threads}"
+        embedding_backend_id = f"{model_name}{cache_dir}{threads}{model_kwargs}"
```
Comment (anakin87) [inline integrations/fastembed/src/haystack_integrations/components/embedders/fastembed/embedding_backend/fastembed_backend.py:78]:
```suggestion
        embedding_backend_id = f"{model_name}{cache_dir}{threads}{local_files_only}{model_kwargs}"
```
It's unrelated to this PR, but `local_files_only` should be part of the id.

File: integrations/fastembed/src/haystack_integrations/components/embedders/fastembed/embedding_backend/fastembed_backend.py [hunk 2]
```diff
@@ -97,9 +102,14 @@ def __init__(
...
         local_files_only: bool = False,
+        model_kwargs: Optional[Dict[str, Any]] = None,
     ):
         self.model = SparseTextEmbedding(
-            model_name=model_name, cache_dir=cache_dir, threads=threads, local_files_only=local_files_only
+            model_name=model_name,
+            cache_dir=cache_dir,
+            threads=threads,
+            local_files_only=local_files_only,
+            **(model_kwargs if model_kwargs else {}),
```
Comment (anakin87) [inline integrations/fastembed/src/haystack_integrations/components/embedders/fastembed/embedding_backend/fastembed_backend.py:112]: for future readibility, let's unwrap this expression

something like:
```python
model_kwargs = model_kwargs or {}

self.model = SparseTextEmbedding(..., **model_kwargs)
```

File: integrations/fastembed/src/haystack_integrations/components/embedders/fastembed/fastembed_sparse_document_embedder.py [hunk 3]
```diff
@@ -81,6 +82,7 @@ def __init__(
         :param local_files_only: If `True`, only use the model files in the `cache_dir`.
         :param meta_fields_to_embed: List of meta fields that should be embedded along with the Document content.
         :param embedding_separator: Separator used to concatenate the meta fields to the Document content.
+        :param model_kwargs: Dictionary containing model parameters such as (`k`, `b`, `avg_len`, `language`).
```
Comment (anakin87) [inline integrations/fastembed/src/haystack_integrations/components/embedders/fastembed/fastembed_sparse_document_embedder.py:85]:
```suggestion
        :param model_kwargs: Dictionary containing model parameters such as `k`, `b`, `avg_len`, `language`.
```

File: integrations/fastembed/src/haystack_integrations/components/embedders/fastembed/fastembed_sparse_text_embedder.py [hunk 4]
```diff
@@ -50,6 +51,7 @@ def __init__(
                 If 0, use all available cores.
                 If None, don't use data-parallel processing, use default onnxruntime threading instead.
         :param local_files_only: If `True`, only use the model files in the `cache_dir`.
+        :param model_kwargs: Dictionary containing model parameters such as (`k`, `b`, `avg_len`, `language`).
```
Comment (anakin87) [inline integrations/fastembed/src/haystack_integrations/components/embedders/fastembed/fastembed_sparse_text_embedder.py:54]:
```suggestion
        :param model_kwargs: Dictionary containing model parameters such as `k`, `b`, `avg_len`, `language`.
```

**Review by anakin87 — COMMENTED** [review]
This PR is almost ready...
I found some minor opportunities for improvement.

**Review by anakin87 — APPROVED** [review]
LGTM!

**Pull request merged**